modules/standard/CTypes.chpl

```diff
@@ -1512,15 +1512,89 @@ module CTypes {
   }
 
   /*
-    Get the number of bytes in a c_ptrConst(int(8)) or c_ptrConst(uint(8)), excluding the
-    terminating null.
+    Get the number of bytes in a c_ptrConst(int(8)) or c_ptrConst(uint(8)),
+    excluding the terminating NULL.
 
     :arg x: c_ptrConst(int(8)) or c_ptrConst(uint(8)) to get length of
 
-    :returns: the number of bytes in x, excluding the terminating null
+    :returns: the number of bytes in x, excluding the terminating NULL
    */
   @unstable("the strLen function is unstable and may change or go away in a future release")
   inline proc strLen(x:c_ptrConst(?t)): int {
      return __primitive("string_length_bytes", x).safeCast(int);
   }
+
+  /*
+    Get a `c_ptrConst(c_char)` from a :type:`~String.string`. The returned
+    `c_ptrConst(c_char)` shares the buffer with the :type:`~String.string`.
+
+    .. warning::
+
+        This can only be called safely on a :type:`~String.string` whose home is
+        the current locale.  This property can be enforced by calling
+        :proc:`~String.string.localize()` before :proc:`string.c_str()`.
+        If the string is remote, the program will halt.
+
+    For example:
+
+    .. code-block:: chapel
+
+      var my_string = "Hello!";
+      on different_locale {
+        printf("%s", my_string.localize().c_str());
```

Comment:
`printf` -> `writef`. And `different_locale`/`my_string` should probably be camel cased.